python/ml4ir/applications/ranking/pipeline.py

```diff
@@ -66,6 +67,23 @@ def get_loss(self):
             output_name=self.args.output_name
         )
 
+    def get_aux_loss(self):
+        """
+        Get the auxiliary loss function to be used with the RelevanceModel
+
+        Returns
+        -------
+        RelevanceLossBase object
+        """
+        if self.aux_loss_key:
+            return loss_factory.get_loss(
+                loss_key=self.aux_loss_key,
+                scoring_type=self.scoring_type,
+                output_name="aux_{}".format(self.args.output_name)
+            )
+        else:
+            return None
```

Comment:
@mohazahran I've also removed the `is_aux_loss` flag in the losses and made them separate classes. You can specify which aux loss you want similar to before with the `aux_loss_key` arg.